src/LinAlg/DiagMatrix.h

```diff
@@ -0,0 +1,115 @@
+// $Id$
+//==============================================================================
+//!
+//! \file DiagMatrix.h
+//!
+//! \date Aug 29 2019
+//!
+//! \author Knut Morten Okstad / SINTEF
+//!
+//! \brief Diagonal system matrix representation.
+//!
+//==============================================================================
+
+#ifndef _DIAG_MATRIX_H
+#define _DIAG_MATRIX_H
+
+#include "SystemMatrix.h"
+
+class SparseMatrix;
```

Comment:
necessary?

Reply:
Not at all, a survivor from DenseMatrix which was used as template here.